runtime/protocol/http-client/common/src/aws/smithy/kotlin/runtime/http/operation/SdkOperationExecution.kt

```diff
@@ -294,6 +294,9 @@ internal class AuthHandler<Input, Output>(
             // update the request context with endpoint specific auth signing context
             val endpointAuthAttributes = endpoint.authOptions.firstOrNull { it.schemeId == authScheme.schemeId }?.attributes ?: emptyAttributes()
             request.context.merge(endpointAuthAttributes)
+
+            // also update the request context with endpoint attributes
+            request.context.merge(endpoint.attributes)
```

Comment:
Without this change, `AttributeKey(backend)=S3Express` is not set in the execution context. Should it be coming in from the `endpointAuthAttributes`? If so, there may be a bug here somewhere

Reply:
I think I see. It would appear we merge the signing attributes which is the extracted endpoint auth attributes for the chose auth scheme. This may be the first case where there are other endpoint attributes unrelated to auth?

It sure feels like this belongs to auth attributes still since it's related to the S3 express auth scheme right? What does the endpoint rule look like here?

Reply:
It looks like this:
```
    "endpoint": {
        "url": "{url#scheme}://{url#authority}/{uri_encoded_bucket}{url#path}",
        "properties": {
            "backend": "S3Express",
            "authSchemes": [{
                "disableDoubleEncoding": true,
                "name": "sigv4",
                "signingName": "s3express",
                "signingRegion": "{Region}"
            }]
        },
        "headers": {}
    }
```